pinot-common/src/main/java/org/apache/pinot/common/function/scalar/DataTypeConversionFunctions.java

```diff
@@ -39,6 +39,12 @@ public class DataTypeConversionFunctions {
   private DataTypeConversionFunctions() {
   }
 
+  // SQL Calcite fails to parse 'cast(value, targetType)' as cast is a reserved SQL keyword.
+  @ScalarFunction
+  public static Object cast2(Object value, String targetTypeLiteral) {
```

Comment:
Let's not add this function. The SQL syntax for cast is `CAST(value AS targetType)`, e.g. `CAST(colA AS BIG_DECIMAL)`

Reply:
@Jackie-Jiang I am using this as ingestion transform. Can I set ingestion transform to "CAST(colName AS targetType)"?